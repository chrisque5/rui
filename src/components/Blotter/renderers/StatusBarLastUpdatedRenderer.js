import { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { getBlotterLastUpdated } from '../../../utils/selectors';
import { formatDate } from '../../../utils/agGrid/index';

export default function StatusBarLastUpdatedRenderer() {
  const blotterLastUpdated = useSelector(getBlotterLastUpdated);
  const [formattedDate, setFormattedDate] = useState();

  useEffect(() => {
    if (blotterLastUpdated !== null) {
      setFormattedDate(`${formatDate(blotterLastUpdated, 'L')} ${formatDate(blotterLastUpdated, 'LTS')}`);
    } else {
      setFormattedDate('N/A');
    }
  }, [blotterLastUpdated]);

  return (
    <div className="ag-name-value">
      <span>Last Updated: </span>
      <span className="ag-name-value-value" data-testid="blotter-last-updated">{formattedDate}</span>
    </div>
  );
}
