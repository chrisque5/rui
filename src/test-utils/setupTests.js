// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
