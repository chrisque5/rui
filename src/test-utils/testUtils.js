/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
