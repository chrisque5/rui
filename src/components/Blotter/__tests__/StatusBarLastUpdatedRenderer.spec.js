import { renderWithProvider } from 'test-utils/testUtils';
import StatusBarLastUpdatedRenderer from '../renderers/StatusBarLastUpdatedRenderer';

describe('<StatusBarLastUpdatedRenderer />', () => {
  test('renders snapshot as expected when no last updated date set in state', () => {
    const initialState = {};

    const { container } = renderWithProvider(
      <StatusBarLastUpdatedRenderer />, { initialState },
    );
    expect(container).toMatchSnapshot();
  });

  test('renders snapshot as expected when last updated date is set in state', () => {
    const initialState = { blotter: { lastUpdated: '12/07/2030 12:34:56' } };

    const { container } = renderWithProvider(<StatusBarLastUpdatedRenderer />, { initialState });
    expect(container).toMatchSnapshot();
  });
});
