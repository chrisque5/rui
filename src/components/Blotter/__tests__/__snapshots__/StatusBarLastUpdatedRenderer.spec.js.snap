// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBarLastUpdatedRenderer /> renders snapshot as expected when last updated date is set in state 1`] = `
<div>
  <div
    class="ag-name-value"
  >
    <span>
      Last Updated: 
    </span>
    <span
      class="ag-name-value-value"
      data-testid="blotter-last-updated"
    >
      12/07/2030 12:34:56 PM
    </span>
  </div>
</div>
`;

exports[`<StatusBarLastUpdatedRenderer /> renders snapshot as expected when no last updated date set in state 1`] = `
<div>
  <div
    class="ag-name-value"
  >
    <span>
      Last Updated: 
    </span>
    <span
      class="ag-name-value-value"
      data-testid="blotter-last-updated"
    >
      N/A
    </span>
  </div>
</div>
`;
